src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/FlowCrudV2Spec.groovy

```diff
@@ -940,6 +940,17 @@ class FlowCrudV2Spec extends HealthCheckSpecification {
             it.source.vlanId == newVlanId
         }
 
+        and: "Flow history shows actual info into stateBefore and stateAfter sections"
+        def flowHistory = northbound.getFlowHistory(flow.flowId)
+        with(flowHistory.last().dumps.find { it.type == "stateBefore" }){
+            it.sourcePort != newPortNumber
+            it.sourceVlan != newVlanId
```

Comment:
it just feels that we should save the old values and do a proper comparison here

Reply:
fixed